### DISCUSSION TITLE
srsue will not stay attached and MCS cannot reach high, with GPSDO equipped

### DISCUSSION THREAD

**Discussion by pengnist** [srsran/srsRAN_Project#554] [General Help]
Hi,

I would like to report a bug that I did not find any mentioning about it.

We have several B210s with and without GPSDOs (TCXO). When connecting the srsue to gnb without the GPSDO, connection seems to be working fine and we can reach downlink MCS as high as 28. 

The issue is with the ones with GPSDO. Although we still configure the sync source to be EXTERNAL, as long as the GPSDO is physically on the B210 board, the downlink throughput can only reach around 18 Mbps, and the DL MCS cannot stay beyond 21. The connection always drops after several minutes while running iperf3 with 30M bandwidth. If I unplug the GPSDO, it could stay connected and MCS can reach 28. 

I am not sure if if the issue is caused by srsRAN Project or the UHD driver, but I think it is worth mentioning here.

Some other information here:
1. I tested powering the B210 with the external power supply while GPSDO is on board, but issue persists.
2. I branched out the srsRAN_Project from the commit bcf941b34faba5e14b4d614772fa45068721afdf (the commits after will always release RRC a few seconds after srsue is connected).

**Comment by xavierarteaga** [comment]
Hi,
first of all, many thanks for your interest.

There are some issues related to the operation of the GPSDO. The problems you see could be related to #548. The patch in [this comment](https://github.com/srsran/srsRAN_Project/issues/548#issuecomment-2044354351) might help. Could you try the patch and give us some feedback?

The patch will be included in the next release.

**Reply by pengnist** [reply]
Thank you so much for the information. I read this post while investigating the issue. The difference is that in #548 the author was using GPSDO as sync and clock, while in our setup, we configured sync source as external and the feed was from an external octoclock. But with GPSDO physically plugged on B210, we could not reach full throughput. Thank you for directing me to the patch file as it may solve both problems. I will try it later this week or earlier next week.

**Reply by pengnist** [reply]
Hi @xavierarteaga , I was able to try the connection with the patch applied. Unfortunately, I was still not able to optimize the MCS. I also tried: 1) gNB USRP with GPSDO + srsUE USRP without GPSDO, or 2) gNB USRP without GPSDO + srsUE USRP with GPSDO,
while the clock and sync sources are both set to external. However, neither would get MCS=28. Looks like as long as GSPDO is on a B210 board, the throughput cannot be optimized.

**Reply by pgawlowicz** [reply]
@pengnist any update on this investigation?
